qiskit/tools/visualization/_state_visualization.py

```diff
@@ -173,79 +221,105 @@ def plot_state_city(rho, title="", filename=None, show=False):
     dzr = datareal.flatten()
     dzi = dataimag.flatten()
 
-    fig = plt.figure(figsize=(8, 8))
-    ax1 = fig.add_subplot(2, 1, 1, projection='3d')
-    ax1.bar3d(xpos, ypos, zpos, dx, dy, dzr, color="g", alpha=0.5)
-    ax2 = fig.add_subplot(2, 1, 2, projection='3d')
-    ax2.bar3d(xpos, ypos, zpos, dx, dy, dzi, color="g", alpha=0.5)
+    if color is None:
+        color = ["#648fff", "#648fff"]
+    else:
+        if len(color) != 2:
+            raise ValueError("'color' must be a list of len=2.")
+        if color[0] is None:
+            color[0] = "#648fff"
+        if color[1] is None:
+            color[1] = "#648fff"
+
+    # set default figure size
+    if figsize is None:
+        figsize = (15, 5)
+
+    fig = plt.figure(figsize=figsize)
+    ax1 = fig.add_subplot(1, 2, 1, projection='3d')
+    ax1.bar3d(xpos, ypos, zpos, dx, dy, dzr, color=color[0], alpha=0.5)
+    ax2 = fig.add_subplot(1, 2, 2, projection='3d')
+    ax2.bar3d(xpos, ypos, zpos, dx, dy, dzi, color=color[1], alpha=0.5)
 
     ax1.set_xticks(np.arange(0.5, lx+0.5, 1))
     ax1.set_yticks(np.arange(0.5, ly+0.5, 1))
-    ax1.axes.set_zlim3d(-1.0, 1.0001)
-    ax1.set_zticks(np.arange(-1, 1, 0.5))
-    ax1.w_xaxis.set_ticklabels(row_names, fontsize=12, rotation=45)
-    ax1.w_yaxis.set_ticklabels(column_names, fontsize=12, rotation=-22.5)
-    # ax1.set_xlabel('basis state', fontsize=12)
-    # ax1.set_ylabel('basis state', fontsize=12)
-    ax1.set_zlabel("Real[rho]")
+    ax1.axes.set_zlim3d(np.min(dzr), np.max(dzr)+1e-9)
+    ax1.zaxis.set_major_locator(MaxNLocator(5))
+    ax1.w_xaxis.set_ticklabels(row_names, fontsize=14, rotation=45)
+    ax1.w_yaxis.set_ticklabels(column_names, fontsize=14, rotation=-22.5)
+    ax1.set_zlabel("Real[rho]", fontsize=14)
+    for tick in ax1.zaxis.get_major_ticks():
+        tick.label.set_fontsize(14)
 
     ax2.set_xticks(np.arange(0.5, lx+0.5, 1))
     ax2.set_yticks(np.arange(0.5, ly+0.5, 1))
-    ax2.axes.set_zlim3d(-1.0, 1.0001)
-    ax2.set_zticks(np.arange(-1, 1, 0.5))
-    ax2.w_xaxis.set_ticklabels(row_names, fontsize=12, rotation=45)
-    ax2.w_yaxis.set_ticklabels(column_names, fontsize=12, rotation=-22.5)
+    if np.min(dzi) != np.max(dzi):
+        eps = 0
+        ax2.zaxis.set_major_locator(MaxNLocator(5))
+    else:
+        ax2.set_zticks([0])
+        eps = 1e-9
+    ax2.axes.set_zlim3d(np.min(dzi), np.max(dzi)+eps)
+    ax2.w_xaxis.set_ticklabels(row_names, fontsize=14, rotation=45)
+    ax2.w_yaxis.set_ticklabels(column_names, fontsize=14, rotation=-22.5)
     # ax2.set_xlabel('basis state', fontsize=12)
     # ax2.set_ylabel('basis state', fontsize=12)
-    ax2.set_zlabel("Imag[rho]")
+    ax2.set_zlabel("Imag[rho]", fontsize=14)
+    for tick in ax2.zaxis.get_major_ticks():
+        tick.label.set_fontsize(14)
     plt.title(title)
```

Comment:
Just like with hinton this is only setting the title over the ax2 plot (it's also a very small fontsize)